### PR TITLE
Control to csv

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

This script takes in a control file and outputs a CSV file for DISA.

#### Rationale:

To improve the process submitting to DISA.

### CODE REVIEW

File: utils/create_srg_export.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        sys.stderr.write(f"Unable to find control file {control_full_path}\n")
+        exit(1)
+    srgs = get_srg_dict(args.manual)
+    control_yaml = ssg.yaml.open_raw(control_full_path)
+    with open(args.json, 'r') as json_file:
+        rule_json = json.load(json_file)
+
+    full_output = pathlib.Path(args.output)
+    with open(full_output, 'w') as csv_file:
+        csv_writer = setup_csv_writer(csv_file)
```
Comment (jan-cerny) [inline utils/create_srg_export.py:132]: It would be better to extract this block to a separate function.

File: utils/create_srg_export.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--control', type=str, action="store", required=True,
+                        help="The control file to parse")
+    parser.add_argument('-o', '--output', type=str, help="The path to the output",
+                        default=OUTPUT)
+    parser.add_argument("-r", "--root", type=str, action="store", default=SSG_ROOT,
+                        help=f"Path to SSG root directory (defaults to {SSG_ROOT})")
+    parser.add_argument("-j", "--json", type=str, action="store", default=RULES_JSON,
+                        help="Path to the rules_dir.json (defaults to build/stig_control.json)")
```
Comment (jan-cerny) [inline utils/create_srg_export.py:114]:
```suggestion
                        help="Path to the rules_dir.json (defaults to {RULES_JSON})")
```

File: utils/create_srg_export.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        row['Artifact Description'] = control.artifact_description
+        row['Status Justification'] = control.status_justification
+        row['Status'] = DisaStatus.from_string(control.status)
+        csv_writer.writerow(row)
+
+
+def create_base_row(item, srgs):
+    row = dict()
+    srg_id = item['id']
+    srg = srgs[srg_id]
```
Comment (jan-cerny) [inline utils/create_srg_export.py:150]: What if the ID isn't present in `srgs`?

Comment (Mab879) [reply utils/create_srg_export.py:153]: The script now errors out in a nice way.

**Review by jan-cerny — COMMENTED** [review]
@Mab879  Unfortunately, with the latest version, I can't generate anything, I only receive an error message.

```
[jcerny@thinkpad scap-security-guide{pr/7775}]$ utils/create_srg_export.py -p rhel8 -c controls/stig_rhel8.yml 
Unable to find SRG RHEL-08-010000

```

What should I do?

**Comment by Mab879** [comment]
@jan-cerny This script only works with controls that SRGs for IDs. Please try with my updates using the RHEL9 STIG control.

**Comment by jan-cerny** [comment]
@Mab879 I would like to merge it, but I can't, because  the incident with Fedora 34 gating job really happened, so please rebase it on master.

**Comment by Mab879** [comment]
@jan-cerny rebase pushed.

**Pull request merged**